include/nbl/video/CThreadSafeGPUQueueAdapter.h

```diff
@@ -34,11 +34,14 @@ class NBL_API CThreadSafeGPUQueueAdapter : public IGPUQueue
             return originalQueue->submit(_count, _submits, _fence);
         }
 
+        // TODO handle this thread safe gpu queue on swapchain present method
```

Comment:
make `CVulkanSwapchain` a friend of `CThreadSafeGPUQueueAdapter` have it lock the queue during the present (you also need to change the signature of `present` to only accept the threadsafe queue)